cmake/OpenCVFindAVIF.cmake

```diff
@@ -0,0 +1,46 @@
+#=============================================================================
+# Find AVIF library
+#=============================================================================
+# Find the native AVIF headers and libraries.
+#
+#  AVIF_INCLUDE_DIRS - where to find avif/avif.h, etc.
+#  AVIF_LIBRARIES    - List of libraries when using AVIF.
+#  AVIF_FOUND        - True if AVIF is found.
+#=============================================================================
+
+# Look for the header file.
+
+unset(AVIF_FOUND)
+
+#find_package(libavif QUIET)
+
+if(TARGET avif)
+    MARK_AS_ADVANCED(AVIF_INCLUDE_DIR)
+    MARK_AS_ADVANCED(AVIF_LIBRARY)
+
+    SET(AVIF_FOUND TRUE)
+    SET(AVIF_LIBRARY avif)
+    GET_TARGET_PROPERTY(AVIF_INCLUDE_DIR1 avif INCLUDE_DIRECTORIES)
+    GET_TARGET_PROPERTY(AVIF_INCLUDE_DIR2 avif INTERFACE_INCLUDE_DIRECTORIES)
+    set(AVIF_INCLUDE_DIR)
+    if(AVIF_INCLUDE_DIR1)
+        LIST(APPEND AVIF_INCLUDE_DIR ${AVIF_INCLUDE_DIR1})
+    endif()
+    if(AVIF_INCLUDE_DIR2)
+        LIST(APPEND AVIF_INCLUDE_DIR ${AVIF_INCLUDE_DIR2})
+    endif()
+else()
```

Comment:
The first branch does not work. Nobody defines avif target. It makes sense to use `AVIF_FOUND`, if  you uncomment find_package.

Reply:
My bad, find_package was commented out for debug ....